lib/dataSource-url.js

```diff
@@ -94,6 +94,9 @@ kettle.dataSource.URL.requestOptions = ["url", "protocol", "host", "hostname", "
 kettle.dataSource.URL.prepareRequestOptions = function (componentOptions, cookieJar, userOptions, permittedOptions, directModel, userStaticOptions, resolveUrl) {
     var staticOptions = fluid.filterKeys(componentOptions, permittedOptions);
     var requestOptions = fluid.extend(true, {headers: {}}, userStaticOptions, staticOptions, userOptions);
+    if (requestOptions.hostname === "localhost" || requestOptions.host === "localhost") {
```

Comment:
Very good - please leave a comment reference to GPII-2147 so future readers can understand the issue